### PR TITLE
Postlesson

### PR DESCRIPTION
**Pull request by VadymPolianskyi**
work after our last lesson
Have corrected routers and handlers, created factory

### CODE REVIEW

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 1]
```diff
@@ -17,16 +17,24 @@
...
-        tRequest.setHeader(msg.getHeader());
-        JacksonDataMapper jacksonDataMapper = new JacksonDataMapper();
-
-        Class thisClass = AbstractRequestHandler.class;
-        ParameterizedType type = (ParameterizedType) thisClass.getGenericSuperclass();
-        Class parameter = (Class) type.getActualTypeArguments()[0];
-        tRequest.setData(jacksonDataMapper.convert((Map<String, Object>)msg.getData(), (Class<T>) parameter));
-
-        return doHandle(tRequest);
+//        Request<T> tRequest = new Request<>();
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:20]: @VadymPolyanski remove commented code

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 2]
```diff
@@ -17,16 +17,24 @@
...
+//        tRequest.setHeader(msg.getHeader());
+//        JacksonDataMapper jacksonDataMapper = new JacksonDataMapper();
+//
+//        Class thisClass = AbstractRequestHandler.class;
+//        ParameterizedType type = (ParameterizedType) thisClass.getGenericSuperclass();
+//        Class parameter = (Class) type.getActualTypeArguments()[0];
+//        tRequest.setData(jacksonDataMapper.convert((Map<String, Object>)msg.getData(), (Class<T>) parameter));
+
+
+        DataMapper mapper = new JacksonDataMapper();
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:30]: @VadymPolyanski Why new data mapper?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 3]
```diff
@@ -17,16 +17,24 @@
...
+        Map<String, Object> map = mapper.convertToMap(msg);
+        RequestData data = mapper.convert(map, RequestData.class);
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:32]: @VadymPolyanski Why you need two conversions? What will you get when trying to convert to interface type ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java [hunk 4]
```diff
@@ -5,7 +5,7 @@
 import com.softgroup.common.router.api.factory.HandlerFactory;
 import com.softgroup.common.router.api.factory.RequestHandlerFactory;
 
-public abstract class AbstractRouterHandler<T extends Handler> implements RouterHandler, CommonRouterHandler {
+public abstract class AbstractRouterHandler<T extends RequestHandler> implements RouterHandler, CommonRouterHandler {
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java:8]: @VadymPolyanski Why request handler? Router can work with other routers same way as with handlers

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java [hunk 5]
```diff
@@ -5,7 +5,7 @@
...
 
 	HandlerFactory<T> factory = new RequestHandlerFactory<T>();
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java:10]: @VadymPolyanski New factory? For what we are learning dependency injection ???

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/impl/FirstRouterHandler.java [hunk 6]
```diff
@@ -18,26 +16,17 @@
...
-        }
-    }
-
 
     @Override
     public String getName() {
         return "mainRouter";
     }
 
+    HandlerFactory<T> factory = new RouterHandlerFactory<T>();
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/impl/FirstRouterHandler.java:25]: @VadymPolyanski Why new factory ???

File: softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/message/CreateConversationRequest.java [hunk 7]
```diff
@@ -10,6 +10,11 @@
  * Time: 19:13
  */
 public class CreateConversationRequest extends MessengerRequest{
+
+    public CreateConversationRequest() {
+        getHeader().setCommand("createConversation");
```
Comment (Dominys) [inline softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/message/CreateConversationRequest.java:15]: @VadymPolyanski Why we need this ?

File: softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/message/DeleteConversationRequest.java [hunk 8]
```diff
@@ -7,6 +7,9 @@
  */
 public class DeleteConversationRequest extends MessengerRequest {
     private String conversationID;
+    public DeleteConversationRequest() {
+        getHeader().setCommand("deleteConversation");
```
Comment (Dominys) [inline softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/message/DeleteConversationRequest.java:11]: @VadymPolyanski ?

File: softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/message/GetConversationByIdsRequest.java [hunk 9]
```diff
@@ -12,6 +12,9 @@
         return conversationIDs;
     }
 
+    public GetConversationByIdsRequest() {
+        getHeader().setCommand("getConversationByIds");
```
Comment (Dominys) [inline softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/message/GetConversationByIdsRequest.java:16]: @VadymPolyanski ?

File: softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/message/GetConversationDetailsRequest.java [hunk 10]
```diff
@@ -8,6 +8,10 @@
 public class GetConversationDetailsRequest extends MessengerRequest {
     private String conversationID;
 
+    public GetConversationDetailsRequest() {
+        getHeader().setCommand("getConversationDetails");
```
Comment (Dominys) [inline softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/message/GetConversationDetailsRequest.java:12]: @VadymPolyanski ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java (hunk 1 above)
Comment (VadymPolianskyi) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:20]: have done

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java (hunk 2 above)
Comment (VadymPolianskyi) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:30]: @Dominys Will @Autowired  be better?

Comment (Dominys) [reply softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:30]: @VadymPolyanski Autowire is the only way! For what we are using Spring?

**Pull request closed without merging**